firestore-next/test.firestore.js

```diff
@@ -1323,3 +1323,1294 @@ describe("firestore", () => {
+            "country": "USA",
+            "state": "California"
+          }
+        });
+        await setDoc(doc(collection(db, "cities"), "TO"), {
+          "name": "Toronto",
+          "population": 3000000,
+          "province": "ON",
+          "location": {
+            "country": "Canada",
+            "province": "Ontario"
+          }
+        });
+        await setDoc(doc(collection(db, "cities"), "NY"), {
+          "name": "New York",
+          "location": {
+            "country": "USA",
+            "state": "New York"
+          }
+        });
+        await setDoc(doc(collection(db, "cities"), "AT"), {
+          "name": "Atlantis",
+        });
+        // [END initial_data]
+
+        // [START full_replace]
+        const names = await execute(db.pipeline()
+          .collection("cities")
+          .replaceWith(field("location"))
+        );
+        // [END full_replace]
+
+        // [START map_merge_overwrite]
+        // unsupported in client SDKs for now
+        // [END map_merge_overwrite]
+        console.log(names);
+    }
+
+    async function sampleStage() {
+        // [START sample_example]
+        let results;
+
+        // Get a sample of 100 documents in a database
+        results = await execute(db.pipeline()
+          .database()
+          .sample(100)
+        );
+
+        // Randomly shuffle a list of 3 documents
+        results = await execute(db.pipeline()
+          .documents([
+            doc(db, "cities", "SF"),
+            doc(db, "cities", "NY"),
+            doc(db, "cities", "DC"),
+          ])
+          .sample(3)
+        );
+        // [END sample_example]
+        console.log(results);
+    }
+
+    async function samplePercent() {
+        // [START sample_percent]
+        // Get a sample of on average 50% of the documents in the database
+        const results = await execute(db.pipeline()
+          .database()
+          .sample({ percentage: 0.5 })
+        );
+        // [END sample_percent]
+        console.log(results);
+    }
+
+    async function unionStage() {
+        // [START union_stage]
+        const results = await execute(db.pipeline()
+          .collection("cities/SF/restaurants")
+          .where(field("type").equal("Chinese"))
+          .union(db.pipeline()
+            .collection("cities/NY/restaurants")
+            .where(field("type").equal("Italian")))
+          .where(field("rating").greaterThanOrEqual(4.5))
+          .sort(field("__name__").descending())
+        );
+        // [END union_stage]
+        console.log(results);
+    }
+
+    async function unionStageStable() {
+        // [START union_stage_stable]
+        const results = await execute(db.pipeline()
+          .collection("cities/SF/restaurants")
+          .where(field("type").equal("Chinese"))
+          .union(db.pipeline()
+              .collection("cities/NY/restaurants")
+              .where(field("type").equal("Italian")) /*, { stable: true } */)
```

Comment:
@MarkDuckworth is `stable` supposed to be in the web API? I couldn't find it on Pipeline anywhere.

Reply:
no. I'm not sure what it would be used for?

Reply:
Seems like this was removed from the doc referenced in the comment above the function at some point. I'll delete it.